webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/users/RudderUserDetailsFile.scala

```diff
@@ -337,50 +340,12 @@ final class FileUserDetailListProvider(roleApiMapping: RoleApiMapping, authorisa
 
 }
 
-object UserFileProcessing {
-
-  val JVM_AUTH_FILE_KEY      = "rudder.authFile"
-  val DEFAULT_AUTH_FILE_NAME = "demo-rudder-users.xml"
-
-  // utility classes for a parsed custom role/user/everything before sanity check is done on them
-  final case class ParsedRole(name: String, permissions: List[String])
-  final case class ParsedUser(name: String, password: String, permissions: List[String], tenants: Option[List[String]])
-  final case class ParsedUserFile(
-      encoder:         PasswordEncoder.Rudder,
-      isCaseSensitive: Boolean,
-      customRoles:     List[UncheckedCustomRole],
-      users:           List[ParsedUser]
-  )
-
-  def getUserResourceFile(): IOResult[UserFile] = {
-    java.lang.System.getProperty(JVM_AUTH_FILE_KEY) match {
-      case null | "" => // use default location in classpath
-        ApplicationLoggerPure.Authz.info(
-          s"JVM property -D${JVM_AUTH_FILE_KEY} is not defined, using configuration file '${DEFAULT_AUTH_FILE_NAME}' in classpath"
-        ) *>
-        UserFile(
-          "classpath:" + DEFAULT_AUTH_FILE_NAME,
-          () => this.getClass().getClassLoader.getResourceAsStream(DEFAULT_AUTH_FILE_NAME)
-        ).succeed
-      case x         => // so, it should be a full path, check it
-        val config = new File(x)
-        if (config.exists && config.canRead) {
-          ApplicationLoggerPure.Authz.info(
-            s"Using configuration file defined by JVM property -D${JVM_AUTH_FILE_KEY} : ${config.getPath}"
-          ) *>
-          UserFile(config.getAbsolutePath, () => new FileInputStream(config)).succeed
-        } else {
-          ApplicationLoggerPure.Authz.error(
-            s"Can not find configuration file specified by JVM property ${JVM_AUTH_FILE_KEY}: ${config.getPath}; aborting"
-          ) *> Unexpected(s"rudder-users configuration file not found at path: '${config.getPath}'").fail
-        }
-    }
-  }
+class UserFileProcessing(bcryptCost: Int) {
```

Comment:
this class is needed to pass the config value because we don't have access to `RudderConfig.RUDDER_BCRYPT_COST` in rudder-rest